src/inc/til/color.h

```diff
@@ -144,16 +144,16 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
             const auto destinationAlpha = destination.a / 255.0f;
             const auto aInverse = 1.0f - sourceAlpha;
 
-            const auto resultA = a + destination.a * aInverse;
+            const auto resultA = sourceAlpha + destination.a * aInverse;
             const auto resultR = (r * sourceAlpha + destination.r * destinationAlpha * aInverse) / resultA;
             const auto resultG = (g * sourceAlpha + destination.g * destinationAlpha * aInverse) / resultA;
             const auto resultB = (b * sourceAlpha + destination.b * destinationAlpha * aInverse) / resultA;
 
             return {
-                static_cast<uint8_t>(resultR + 0.5f),
-                static_cast<uint8_t>(resultG + 0.5f),
-                static_cast<uint8_t>(resultB + 0.5f),
-                static_cast<uint8_t>(resultA + 0.5f),
+                static_cast<uint8_t>(255.0f * (resultR + 0.5f)),

```

Comment:
_what_? and this doesn't break the other uses of `layer_over`?

Reply:
Actually this particular line has a bug: `255.0f * (resultR + 0.5f)` is `255.0f * resultR + 255.0f * 0.5f` and thus it always adds `128` to the result.
I believe the `resultA` calculation has a bug as well: `sourceAlpha + destination.a * aInverse` is neither in the range [0, 1] nor in [0, 255] because `sourceAlpha` and `aInverse` are the former and `destination.a` is the latter.